File: dsa/array/121_best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyInd = 0;
        int maxProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] >= prices[buyInd]) {
                int tempProfit = prices[i] - prices[buyInd];
                if (tempProfit > maxProfit) {
                    maxProfit = tempProfit;
                }
            }
            else {
                buyInd = i;
            }
        }
        return maxProfit;
    }
}

File: dsa/array/121_Best_Time_to_Buy_and_Sell_Stock.py
from typing import List
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit: int = 0
        buyInd: int = 0
        ind: int = 0
        while ind < len(prices):
            profit: int = prices[ind] - prices[buyInd]
            if profit > 0:
                maxProfit = max(maxProfit, profit)
            else:
                buyInd = ind
            ind+=1
        return maxProfit